internal/ingress/controller/controller.go

```diff
@@ -231,12 +235,21 @@ type KongController struct {
 	PluginSchemaStore PluginSchemaStore
 
 	Logger logrus.FieldLogger
+
+	tmpDir string
 }
 
 // Start starts a new master process running in foreground, blocking until the next call to
 // Stop.
 func (n *KongController) Start() {
 	n.Logger.Debugf("startin up controller")
+	if len(n.cfg.DumpConfig) > 0 {
+		var err error
+		n.tmpDir, err = ioutil.TempDir("", "controller")
+		if err != nil {
+			panic(fmt.Errorf("failed to create a temporary working directory: %v", err))
```

Comment:
the codebase uses logger's `Fatalf` for this purpose. I recommend changing to that (instead of panicking) for consistency.

Also, `Start()` doesn't seem designed for failures (and it's not easy to cleanly return an error from here) - which reinforces my point about moving this to `main()`.